google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/TableDefinition.java

```diff
@@ -85,6 +85,8 @@ public Type apply(String constant) {
 
     public static final Type SNAPSHOT = type.createAndRegister("SNAPSHOT");
 
+    public static final Type CLONE = type.createAndRegister("CLONE");
```

Comment:
Oh, here's the issue.  We likely shouldn't make a new CloneTableDefinition at all.  The type returned by the table metadata for a clone tables is still TABLE, not CLONE.  It just has the CloneTableDefinition present.

Reply:
Got it.. That Makes a lot more sense. By that definition shouldn't SNAPSHOT also follow a similar implementation flow? Or is that different?
Does creating a SNAPSHOT table change its definition?